### PR TITLE
New config issue format

### PR DESCRIPTION
**Pull request by s0b0lev**
I will copy open question here:

I am trying to configure checker for Github SWC-registry and I wanna generate keccak256 on javascript, 
I use this library: http://emn178.github.io/online-tools/keccak_256.html but it produce different output

We got bytecode string:

> 608060405234801561001057600080fd5b5061010c806100206000396000f300608060405260043610603f576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff1680632553e8a014604d575b348015604a57600080fd5b50005b348015605857600080fd5b50605f6061565b005b60008060008040925082604051808260001916600019168152602001915050604051809103902091505a90506000811415609a57600080fd5b3073ffffffffffffffffffffffffffffffffffffffff166040516000604051808303816000865af291505050600073ffffffffffffffffffffffffffffffffffffffff16ff00a165627a7a72305820602d9137348e21d7141640e6dfe4b67ded3f3b4898e778a72e3464f51aa9cc630029

Which will be translated to (using http://emn178.github.io/online-tools/keccak_256.html):
`eb7bbb275df9c8d7d90cf7bf2204028498a23a9c31e58dec1227d8cd34dccecd`

While our config define it as: 
`bdb45e04caea963dec0c156b81632b1cb8970c6dd9cbacbb2bfc04770dd73895`

How I get this results:
```python
import _pysha3 as sha3

bytecode = "608060405234801561001057600080fd5b5061010c806100206000396000f300608060405260043610603f576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff1680632553e8a014604d575b348015604a57600080fd5b50005b348015605857600080fd5b50605f6061565b005b60008060008040925082604051808260001916600019168152602001915050604051809103902091505a90506000811415609a57600080fd5b3073ffffffffffffffffffffffffffffffffffffffff166040516000604051808303816000865af291505050600073ffffffffffffffffffffffffffffffffffffffff16ff00a165627a7a72305820602d9137348e21d7141640e6dfe4b67ded3f3b4898e778a72e3464f51aa9cc630029"

keccak = sha3.keccak_256()
keccak.update(bytes.fromhex(bytecode))
print(f"0x{keccak.hexdigest()}")
```
`[Output] => "0xbdb45e04caea963dec0c156b81632b1cb8970c6dd9cbacbb2bfc04770dd73895"`

How Valentin get this results in Go:
```golang
func Test1(t *testing.T) {
    decInput, _ := hex.DecodeString("608060405234801561001057600080fd5b5061010c806100206000396000f300608060405260043610603f576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff1680632553e8a014604d575b348015604a57600080fd5b50005b348015605857600080fd5b50605f6061565b005b60008060008040925082604051808260001916600019168152602001915050604051809103902091505a90506000811415609a57600080fd5b3073ffffffffffffffffffffffffffffffffffffffff166040516000604051808303816000865af291505050600073ffffffffffffffffffffffffffffffffffffffff16ff00a165627a7a72305820602d9137348e21d7141640e6dfe4b67ded3f3b4898e778a72e3464f51aa9cc630029")
    hash := crypto.Keccak256Hash(decInput)
    fmt.Printf("0x%x\n", hash)
}
```
`[Output] => "0xbdb45e04caea963dec0c156b81632b1cb8970c6dd9cbacbb2bfc04770dd73895"`

But this UI show different results: http://emn178.github.io/online-tools/keccak_256.html

Whats wrong here @thec00n @wuestholz ?

### CODE REVIEW

**Comment by thec00n** [comment]
For JS you can use Web3 -> https://web3js.readthedocs.io/en/1.0/web3-utils.html

**Comment by s0b0lev** [comment]
There are 2 configs where `bin` key is empty for first contract:

https://github.com/SmartContractSecurity/SWC-registry/blob/master/test_cases/assert_violations/runtime_user_input_call.json

I change script to take first contract with `bin`

File: scripts/yaml_validate.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 const fs = require('fs');
 const path = require('path');
 const yaml = require('js-yaml');
+const Web3 = require('web3');
+
+var web3 = new Web3();
```
Comment (thec00n) [inline scripts/yaml_validate.js]: const web3 = require('web3');

Comment (thec00n) [inline scripts/yaml_validate.js]: remove

Comment (s0b0lev) [reply scripts/yaml_validate.js]: Done

Comment (s0b0lev) [reply scripts/yaml_validate.js]: done

File: test_cases/integer_overflow_and_underflow/integer_overflow_benign_1.yaml [hunk 2]
```diff
@@ -1,7 +1,8 @@
-description: "Single transaction overflow"
+description: Single transaction overflow
 issues:
-- id: "SWC-101"
+- id: SWC-101
   count: 1
   locations:
-  - bytecode_offsets: []
-    line_numbers: [10]
+  - bytecode_offsets: {}
```
Comment (wuestholz) [inline test_cases/integer_overflow_and_underflow/integer_overflow_benign_1.yaml]: 173:1:0 (0x6eac85a0077d7fbe4a212397613039dd0303f5729738128c6248e3a25c54cf46)

File: test_cases/integer_overflow_and_underflow/integer_overflow_minimal.yaml [hunk 3]
```diff
@@ -1,7 +1,8 @@
-description: "Single transaction overflow"
+description: Single transaction overflow
 issues:
-- id: "SWC-101"
+- id: SWC-101
   count: 1
   locations:
-  - bytecode_offsets: []
-    line_numbers: [10]
+  - bytecode_offsets: {}
```
Comment (wuestholz) [inline test_cases/integer_overflow_and_underflow/integer_overflow_minimal.yaml]: 174:1:0 (0x80074413148835f7b77d5bf3607abd9e7cdd53db154097aa0e82b9a7bdfbbfe3)

File: test_cases/integer_overflow_and_underflow/integer_overflow_mapping_sym_1.yaml [hunk 4]
```diff
@@ -1,7 +1,8 @@
-description: "Single transaction overflow in mapping"
+description: Single transaction overflow in mapping
 issues:
-- id: "SWC-101"
+- id: SWC-101
   count: 1
   locations:
-  - bytecode_offsets: []
-    line_numbers: [9]
+  - bytecode_offsets: {}
```
Comment (wuestholz) [inline test_cases/integer_overflow_and_underflow/integer_overflow_mapping_sym_1.yaml]: 145:1:0 (0x3c60415e40f53b65bb32a782b5ca77cb5fdf3e2bbd5e942ce263cd674f02444c)

File: test_cases/integer_overflow_and_underflow/integer_overflow_mul.yaml [hunk 5]
```diff
@@ -1,7 +1,8 @@
-description: "Single transaction overflow using mul"
+description: Single transaction overflow using mul
 issues:
-- id: "SWC-101"
+- id: SWC-101
   count: 1
   locations:
-  - bytecode_offsets: []
-    line_numbers: [10]
+  - bytecode_offsets: {}
```
Comment (wuestholz) [inline test_cases/integer_overflow_and_underflow/integer_overflow_mul.yaml]: 174:1:0 (0x6447009c99f794580933d06bda6405b9d6c4e95c4b51a37bf3c8b36c9199b0cf)

File: test_cases/integer_overflow_and_underflow/overflow_simple_add.yaml [hunk 6]
```diff
@@ -1,7 +1,8 @@
-description: "Plain and simple ADD overflow example"
+description: Plain and simple ADD overflow example
 issues:
-- id: "SWC-101"
+- id: SWC-101
   count: 1
   locations:
-  - bytecode_offsets: []
-    line_numbers: [7]
+  - bytecode_offsets: {}
```
Comment (wuestholz) [inline test_cases/integer_overflow_and_underflow/overflow_simple_add.yaml]: 168:1:0 (0x75ad68f906456e1cbfd6190a8f2e2dc5cb2794af4a4929448378642c992e151a)

File: test_cases/write_to_arbitrary_storage_location/mapping_write.yaml [hunk 7]
```diff
@@ -1,7 +1,8 @@
-description: "Write to arbitrary storage location using dynamic arrays"
+description: Write to arbitrary storage location using dynamic arrays
 issues:
-- id: "SWC-124"
+- id: SWC-124
   count: 1
   locations:
-  - bytecode_offsets: []
-    line_numbers: [14]
+  - bytecode_offsets: {}
```
Comment (wuestholz) [inline test_cases/write_to_arbitrary_storage_location/mapping_write.yaml]: 395:1:0 (0xb3f8b66f8449fff6ee9ba17ae5534dfb2d8f4c4281c8a9ad56c1354a55c389d3)

File: test_cases/arbitrary_jump/FunctionTypes.yaml [hunk 8]
```diff
@@ -1,7 +1,8 @@
-description: "Arbitrary jump with function type variable"
+description: Arbitrary jump with function type variable
 issues:
-- id: "SWC-127"
+- id: SWC-127
   count: 1
   locations:
-  - bytecode_offsets: []
-    line_numbers: [26]
+  - bytecode_offsets: {}
```
Comment (wuestholz) [inline test_cases/arbitrary_jump/FunctionTypes.yaml]: 264:1:0 (0x10262f9b836c8ce4a7c2807a00b08e278094414ceeb2b5cbf4365c7ced0fd2f9)

File: test_cases/assert_violations/gas_model.yaml [hunk 9]
```diff
@@ -1,8 +1,9 @@
...
 issues:
-- id: "SWC-110"
+- id: SWC-110
   count: 1
   locations:
-  - bytecode_offsets: [112]
-    line_numbers: [13]
-
+  - bytecode_offsets:
+      '0x1d0245bf42c590285bb4751565a6fabbdb6825e95c009d8e5392b63d9851d9ab': [112]
```
[the remark this answers is not in the document]
Comment (thec00n) [reply test_cases/assert_violations/gas_model.yaml]: yes

File: test_cases/arbitrary_jump/FunctionTypes.yaml (hunk 8 above)
Comment (thec00n) [reply test_cases/arbitrary_jump/FunctionTypes.yaml]: There is probably multiple locations that could be considered valid in this test case.

File: test_cases/assert_violations/runtime_create_user_input.yaml [hunk 10]
```diff
@@ -1,8 +1,9 @@
...
 issues:
-- id: "SWC-110"
+- id: SWC-110
   count: 1
   locations:
-  - bytecode_offsets: [296]
-    line_numbers: [11]
-
+  - bytecode_offsets:
+      '0x8b3686fb2c2aa036f95720baaa2cc9c033c4b0a9611870d8db0a84aca15d9a27': [296]
```
Comment (wuestholz) [inline test_cases/assert_violations/runtime_create_user_input.yaml]: I think this should be the following instead:
336:1:0 (0x1410f656e6c549fa79b9b13b559ef15eb4149c80d4f22a314642993add450e19)

Comment (wuestholz) [reply test_cases/assert_violations/runtime_create_user_input.yaml]: @s0b0lev @thec00n BTW, does Edelweiss run on each contract separately?

File: test_cases/assert_violations/constructor_create_argument.yaml [hunk 11]
```diff
@@ -1,8 +1,9 @@
...
 issues:
-- id: "SWC-110"
+- id: SWC-110
   count: 1
   locations:
-  - bytecode_offsets: [277]
-    line_numbers: [11]
-
+  - bytecode_offsets:
+      '0x05ca14f58db1545d83289b13561ace31ca7bee82f9e7a1b0c931d415dcb3ddf7': [277]
```
Comment (wuestholz) [inline test_cases/assert_violations/constructor_create_argument.yaml]: I think this should be the following instead:
295:1:0 (0xa9bafdd71a4744d05c44b4cf2f75e05bb66b0252ae3ec370587e49b15fde453c)

File: test_cases/assert_violations/constructor_create.yaml [hunk 12]
```diff
@@ -1,8 +1,9 @@
...
 issues:
-- id: "SWC-110"
+- id: SWC-110
   count: 1
   locations:
-  - bytecode_offsets: [277]
-    line_numbers: [12]
-
+  - bytecode_offsets:
+      '0xfd73371b884f20a50acfd8f790496e905a7c397d16cf5a30dd5fcb461499d470': [277]
```
Comment (wuestholz) [inline test_cases/assert_violations/constructor_create.yaml]: I think this should be the following instead:
295:1:0 (0xc27f0d2e53876fb4e1100569b750ff701f48b85487e89782f8b8e4dd72369e67)

File: test_cases/assert_violations/runtime_create_user_input.yaml (hunk 10 above)
Comment (thec00n) [reply test_cases/assert_violations/runtime_create_user_input.yaml]: yes

**Pull request merged**